pages/k8s/monitoring.md

```diff
@@ -0,0 +1,257 @@
+juju add-relation kubernetes-master:juju-info telegraf:juju-info
+juju add-relation kubernetes-worker:juju-info telegraf:juju-info
+```
+
+### Adding a scraper for Prometheus
+
+Prometheus will also need an appropriate scraper to collect metrics relevant to
+the cluster.  A useful default is installed when using **conjure-up** (the
+template for this can be [downloaded here][download-scraper]), but you can also
+configure it manually by following the steps outlined here:
+
+ -  Download the scraper file
+    ```bash
+    curl -O  https://raw.githubusercontent.com/conjure-up/spells/master/canonical-kubernetes/addons/prometheus/steps/01_install-prometheus/prometheus-scrape-k8s.yaml
+    ```
+    This is the template, but it needs some specific information for your cluster.
+ -  Get the relevant address and password form your cluster
+     ```bash
+    api=$(juju run  --unit kubeapi-load-balancer/0 'network-get website --format yaml --ingress-address' | head -1)
+    pass=$(grep 'password:' ~/.kube/config | sed -e 's/ *//g' -e 's/password://')
+    ```
+    This will fetch the relevant info and store in temporary environment variables for convenience.
+ -   Substitute in the variables
+     ```bash
+     sed -e 's/K8S_PASSWORD/'"$pass"'/' -e 's/K8S_API_ENDPOINT/'"$api"'/' <prometheus-scrape-k8s.yaml  > myscraper.yaml
+     ```
+ -   Configure Prometheus to use this scraper
+     ```bash
+     juju config prometheus scrape-jobs="$(<myscraper.yaml)"
+     ```
+
+### Add the sample dashboard
+
+As with the scraper, there is a [sample dashboard available for download
+here][download-dashboard]. You can download and configure **grafana** to use it
+by following these steps:
+
+ -  Download the sample dashboard configuration
+    ```bash
+    curl -O https://raw.githubusercontent.com/conjure-up/spells/master/canonical-kubernetes/addons/prometheus/steps/01_install-prometheus/grafana-k8s.json
+    ```
+ -  Upload this to **grafana**
+    ```bash
+    juju run-action --wait grafana/0 import-dashboard  dashboard="$(base64 grafana-k8s.json)"
+    ```
+
+### Retrieve  credentials and login
+
+To open the dashboard in your browser you will need to know the IP address for
+**grafana** and the admin password. These can be retrieved with the following
+commands:
+
+```bash
+juju status --format yaml grafana/0 | grep public-address   
+```
+Will return the accessible IP address for the dashboard.
+
+```bash
+juju run-action --wait grafana/0 get-admin-password
+```
+Will return the password for the user 'admin'
+
+You can now navigate to the website at `http://<your-ip>:3000`  and login with
+the username `admin` and the password you just retrieved.
+
+Once logged in, check out the cluster metric dashboard by clicking the `Home`
+drop-down box and selecting `Kubernetes Metrics (via Prometheus)`:
+
+![grafana dashboard image][grafana-1]
+
+You can also check out the system metrics of the cluster by switching the
+drop-down box to `Node Metrics (via Telegraf):
```

Comment:
Probably worth noting that the Telegraf dashboard may not be installed (it is with conjure-up, but not if users are doing these steps manually).

You may want to add another blurb to the **Add the sample dashboard** section above to do the same thing you did for `grafana-k8s.json`, except using `grafana-telegraf.json` instead:

```bash
curl -O https://raw.githubusercontent.com/conjure-up/spells/master/canonical-kubernetes/addons/prometheus/steps/01_install-prometheus/grafana-telegraf.json
juju run-action --wait grafana/0 import-dashboard  dashboard="$(base64 grafana-telegraf.json)"
```

Reply:
done